llvm/lib/ProfileData/InstrProfWriter.cpp

```diff
@@ -350,6 +350,31 @@ bool InstrProfWriter::addMemProfCallStack(
   return true;
 }
 
+bool InstrProfWriter::addMemProfData(memprof::IndexedMemProfData Incoming,
+                                     function_ref<void(Error)> Warn) {
+  if (MemProfData.Frames.empty())
```

Comment:
Should it be the case that either all 3 are empty or none are? If so, add an error check?